scripts/build-cmake.sh

```diff
@@ -4,36 +4,54 @@ set -e
 
 # Build everything with Conan and CMake
 
-BUILDDIR=Build
 BUILDTYPE=Release               # "Release" or "Debug"
 
-if command -v ninja; then
-    DEFAULT_GENERATOR="Ninja"
-elif command -v xcodebuild; then
-    DEFAULT_GENERATOR="Xcode"
-else
-    DEFAULT_GENERATOR="Unix Makefiles"
-fi
 if [[ $1 = "-v" ]]; then
     VERBOSE="--verbose"; shift
 fi
-GENERATOR=${1:-"$DEFAULT_GENERATOR"}      # "Ninja", "Unix Makefiles", "Xcode"
 
-echo "Building OpenFX $BUILDTYPE in $BUILDDIR with $GENERATOR"
+if [[ -n ${1:-} ]]; then
+    GENERATOR=$1
+    GENERATOR_OPTION="-c tools.cmake.cmaketoolchain:generator=${GENERATOR}"
+fi
+
+echo "Building OpenFX $BUILDTYPE in build/ with ${GENERATOR:-conan platform default generator}"
+
+
+CONAN_VERSION=$(conan -v | sed -e 's/Conan version //g')
+CONAN_MAJOR_VERSION=${CONAN_VERSION:0:1}
+
+PRESET_NAME=
+PRESET_BASE_NAME=
+CMAKE_BUILD_DIR=build
+OS=$(uname -s)
+
+if [[ ${OS:0:5} == "MINGW" && ${GENERATOR,,} != "ninja" ]]; then
+    PRESET_BASE_NAME=default
+else
+    PRESET_BASE_NAME=${BUILDTYPE,,}
+    CMAKE_BUILD_DIR=${CMAKE_BUILD_DIR}/${BUILDTYPE}
+fi
+
+if [[ ${CONAN_MAJOR_VERSION} == "1" ]]; then
+    PRESET_NAME=${PRESET_BASE_NAME}
+elif [[ ${CONAN_MAJOR_VERSION} == "2" ]]; then
+    PRESET_NAME=conan-${PRESET_BASE_NAME}
+else
+    echo "Unexpected conan version ${CONAN_VERSION}"
+    exit 1
+fi
 
 set -x
-# Install dependencies and set up build dir
-conan install -if $BUILDDIR -of $BUILDDIR -s build_type="$BUILDTYPE" .
+
+# Install dependencies, set up build dir, and generate build files.
+conan install ${GENERATOR_OPTION} -s build_type=$BUILDTYPE --build=missing .
 
 # Generate the build files
-cmake -S. -B$BUILDDIR \
-      -G"$GENERATOR" \
-      -DBUILD_EXAMPLE_PLUGINS=TRUE \
-      -DCMAKE_BUILD_TYPE=$BUILDTYPE \
-      -DCMAKE_EXPORT_COMPILE_COMMANDS=1
+cmake --preset ${PRESET_NAME} -DBUILD_EXAMPLE_PLUGINS=TRUE
 
 # Do the build
-cmake --build $BUILDDIR $VERBOSE
+cmake --build ${CMAKE_BUILD_DIR} --config $BUILDTYPE $VERBOSE
 
-# install (or you could just do "cmake --build $BUILDDIR --target install")
-cmake --install $BUILDDIR
+# install (or you could just do "cmake --build $BUILDDIR --config $BUILDTYPE --target install")
+#cmake --install build/ --config $BUILDTYPE
```

Comment:
With this line commented out, obviously the plug-ins don't get installed anymore. It should be changed to `cmake --install ${CMAKE_BUILD_DIR} --config $BUILDTYPE`. The comment above about the alternative command should also be changed to incorporate ${CMAKE_BUILD_DIR} instead of $BUILDDIR due to other changes in the script.

Reply:
Apart from this, everything looks good, @garyo if you can make this change I will complete the review.

Reply:
I think @acolwell commented this out because it'll fail the build unless the OFX install dir exists and is world-writable. When I create the CI build script, I can make sure that exists before running this script, but I think for users we should just tell them to run the `cmake --install` command as root (or admin, for Windows). Or do you think we should just fail the build there and add an error message telling them to create the install dir and ensure it's writable by them? Or perhaps check for this up front?

I agree about fixing the build dir name.

Reply:
For a developer new to the project, having the install phase run automatically is very helpful. I would rather see the build fail with a clear permissions error than skip it and leave that new developer scratching their head about how to get the plug-ins to show up in a host. Checking up front would get my vote.

Reply:
Yes. I disabled it because it failed and I had concerns about it writing to a global directory like this. I also disagree with combining the build and install steps here.  "make" then "make install" is a common pattern in projects and allows the developer to explicitly decide whether to install the package in a default location or not.

As a new developer to this project, I was quite surprised that this script was trying to install into a global location. Obviously this is your project and you can do what you will. I just figured I'd respond with my reasoning in the hopes that it would be helpful.

Reply:
I agree "make" then "make install" is well understood. Maybe the thing to do is remove the install step from build-cmake.sh (or enable it via a command-line arg) then document the cmake --install command in readme.md.